Formula/v/veilid.rb

```diff
@@ -0,0 +1,24 @@
+class Veilid < Formula
+  desc "Peer-to-peer network for easily sharing various kinds of data"
+  homepage "https://veilid.com/"
+  url "https://gitlab.com/veilid/veilid/-/archive/v0.2.5/veilid-v0.2.5.tar.gz"
+  sha256 "167c9a140aadc69d02a292d79edf949027d70a02985a860f0068adef914341df"
+  license "MPL-2.0"
+  head "https://gitlab.com/veilid/veilid.git", branch: "main"
+
+  # TODO: Remove `capnp` dependency once version >v0.2.5
+  depends_on "capnp" => :build
+  # TODO: Remove `protobuf` dependency once version >v0.2.5
+  depends_on "protobuf" => :build
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "veilid-cli")
+    system "cargo", "install", *std_cargo_args(path: "veilid-server")
+  end
+
+  test do
+    assert_match "veilid-cli", shell_output(bin/"veilid-cli --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Updated! Thanks for the quick feedback.